web/src/components/Terminal.tsx

```diff
@@ -0,0 +1,480 @@
+import clsx from 'clsx'
+import { useEffect, useRef, useState } from 'react'
+
+export default () => {
+    return (
+        <div className='w-full flex flex-col bg-gray-900 rounded-2xl overflow-hidden relative'>
+            <Header />
+            <Content />
+        </div>
+    )
+}
+
+const Header = () => {
+    const colors = [
+        'bg-red-500',
+        'bg-yellow-500',
+        'bg-green-500',
+    ]
+    return (
+        <div className='flex w-full border-b border-b-slate-800 bg-gray-900 '>
+            {colors.map(color => (
+                <div
+                    className={clsx(
+                        'h-3 w-3 ml-2 my-2 rounded-full',
+                        color
+                    )}
+                    key={color}
+                ></div>
+            ))}
+        </div>
+    )
+}
+
+const lsOutput = `
+total 42
+drwxr-xr-x   27 git-man  staff    864 Feb  8 21:18 .
+drwxr-xr-x   25 git-man  staff    800 Feb 15 18:41 ..
+drwxr-xr-x   16 git-man  staff    512 Feb 14 14:06 .git
+-rw-r--r--    1 git-man  staff   1408 Feb  6 21:44 .gitignore
+-rw-r--r--    1 git-man  staff    154 Nov 13 12:03 code.tsx
+`
+
+const gitCommitOutput = `
+[main 3e6f0d6] Minor change to language.
+1 file changed, 1 insertions(+), 0 deletions(-)`
+
+const gitPushOutput = `
+Enumerating objects: 6, done.
+Counting objects: 100% (6/6), done.
+Delta compression using up to 4 threads
+Compressing objects: 100% (4/4), done.
+Writing objects: 100% (4/4), 481 bytes | 481.00 KiB/s, done.
+Total 4 (delta 2), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (2/2), completed with 1 local object.
+To https://github.com/gibsonbailey/git-streak-tracker.git
+   3e6f0d6..89abcdef  main -> main`
+
+const Content = () => {
+    const [ mode, setMode ] = useState<'shell' | 'vim'>('shell')
+    const [ linesChanged, setLinesChanged ] = useState(0)
+    const [ currentCommand, setCurrentCommand ] = useState('')
+    const [ commandOutputs, setCommandOutputs ] = useState<string[]>([])
+    const [ currentCommandCursorIndex, setCurrentCommandIndex ] = useState(0)
+    const [ commandSpeed, setCommandSpeed ] = useState<'low' | 'high'>('low')
+
+
+    useEffect(() => {
+        setTimeout(() => {
+            setCurrentCommand('ls -la')
+        }, 1500)
+    }, [])
+
+    const vimModeFinished = () => {
+        setMode('shell')
+        setTimeout(() => {
+            setCommandSpeed('high')
+            setCurrentCommand('git commit -m "Minor change to language."')
+            setCurrentCommandIndex(0)
+        }, 1000)
+    }
+
+    useEffect(() => {
+        if (currentCommand == '') {
+            return
+        }
+        if (currentCommandCursorIndex < currentCommand.length) {
+            const timeout = setTimeout(() => {
+                setCurrentCommandIndex(currentCommandCursorIndex + 1)
+            }, Math.floor((commandSpeed === 'low' ? 50 : 20) + (Math.random() * 30)))
+            return () => { clearTimeout(timeout) }
```

Comment:
For ease of reading -- nitpick for sure

```suggestion
            const timeout = setTimeout(() => {
                setCurrentCommandIndex(currentCommandCursorIndex + 1)
            },
                Math.floor((commandSpeed === 'low' ? 50 : 20) + (Math.random() * 30))
            )
            
            return () => { clearTimeout(timeout) }
```

Reply:
I split this up a bit to make it more readable in a recent commit.